tests/platform_helper/domain/test_pipelines.py

```diff
@@ -122,6 +122,29 @@ def test_generate_pipeline_command_generate_terraform_files_for_environment_pipe
     )
 
 
+@freeze_time("2024-10-28 12:00:00")
+@patch("dbt_platform_helper.jinja2_tags.version", new=Mock(return_value="v0.1-TEST"))
+def test_generate_pipeline_generates_expected_terraform_manifest_when_no_deploy_repository_key(
+    fakefs,
+    platform_config_for_env_pipelines,
+):
+
+    app_name = "test-app"
+    # deploy_repository key set on test_fixture so remove it
+    platform_config_for_env_pipelines.pop("deploy_repository")
+    fakefs.create_file(PLATFORM_CONFIG_FILE, contents=yaml.dump(platform_config_for_env_pipelines))
+    mocks = PipelineMocks(app_name)
+    pipelines = Pipelines(**mocks.params())
+
+    pipelines.generate("an-unimportant-verstion", "abranch")
+
+    expected_files_dir = Path(f"terraform/environment-pipelines/platform-prod-test/main.tf")
+    assert expected_files_dir.exists()
+    content = expected_files_dir.read_text()
+
+    assert 'repository          = "uktrade/test-app-deploy"' in content
```

Comment:
Those spaces will make this brittle. Not now, but let's convert this manifest to JSON too. Otherwise every time `terraform fmt` changes the spacing, tests are going to brake seemingly randomly.